External/cmake/CMakeLists.txt

```diff
@@ -0,0 +1,74 @@
+# CMakeLists.txt
```

Comment:
This was a supproject that build dependencies. I feel strongly that in a cross platform application (especially an open source one) that each platform shares the same revisions for it's dependencies and ways to compile the current dependencies should be available in the project. This way no one commits an updated library for one platform with no documentation on how they built it (thus splitting the dependencies and destroying the holistic consistency of the tool). Tools should do their best to work the same on all supported platforms.

Reply:
@UebelAndre Nice work on this, The cmake setup was updated to include external common folder, however it was done for Windows OS, Linux and Mac builds are broken and can use some assistance on fixes. Still relying on VS for lib builds for all configurations, and found issues with installer packages for the libs and sample code. This will need to be fixed and moved to all cmake.